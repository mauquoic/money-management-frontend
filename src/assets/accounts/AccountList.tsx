import React from "react";
import {Account} from "../../models/Account";
import {AccountItem} from "./AccountItem";
import classes from "./AccountList.module.css";
import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import TableBody from "@material-ui/core/TableBody";
import Paper from "@material-ui/core/Paper";

function AccountTableList() {
  const accounts: Array<Account> = [{
    id: 1,
    amount: 320.8,
    currency: "CHF",
    name: "myNiceAccount",
    description: "Wow, this is advancing!"
  }, {
    id: 2,
    amount: 94.6,
    currency: "EUR",
    name: "eurozz",
  }];

  return (
    <div>
      <Paper elevation={6} className={classes.root}>
        <TableContainer className={classes.container}>
          <Table stickyHeader aria-label="sticky table">
            <TableHead>
              <TableRow>
                <TableCell align={'center'}>
                  Name
                </TableCell>
                <TableCell align={'center'}>
                  Value
                </TableCell>
                <TableCell align={'center'}>
                  Description
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {accounts.map((account) => {
                return (
                  <AccountItem key={account.id} account={account}/>
                );
              })}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </div>
  )
}

export default AccountTableList;