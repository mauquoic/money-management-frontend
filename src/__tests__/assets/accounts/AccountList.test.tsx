import React from "react";
import {cleanup, render} from "@testing-library/react";
import AccountList from "../../../assets/accounts/AccountList";

const accounts = [{
  "id": 1,
  "name": "myAccount",
  "description": "This is a description.",
  "currency": "CHF",
  "amount": 100
},
  {
    "id": 2,
    "name": "myAccount2",
    "currency": "USD",
    "amount": 200
  }];

afterEach(cleanup);

describe("AccountList rendering", () => {
  test("AccountList is rendered correctly", () => {
    const accountList = render(<AccountList accounts={accounts}/>);
    expect(accountList).toMatchSnapshot();
  });
});