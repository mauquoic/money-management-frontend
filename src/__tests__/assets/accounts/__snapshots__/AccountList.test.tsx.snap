// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList rendering AccountList is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
      >
        <div
          class="MuiPaper-root root MuiPaper-elevation6 MuiPaper-rounded"
        >
          <div
            class="MuiTableContainer-root container"
          >
            <table
              aria-label="sticky table"
              class="MuiTable-root MuiTable-stickyHeader"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                    scope="col"
                  >
                    Value
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                    scope="col"
                  >
                    Description
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-hover"
                  role="checkbox"
                  tabindex="-1"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    myAccount
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    CHF
                     
                    100
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    This is a description.
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root MuiTableRow-hover"
                  role="checkbox"
                  tabindex="-1"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    myAccount2
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    USD
                     
                    200
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
    >
      <div
        class="MuiPaper-root root MuiPaper-elevation6 MuiPaper-rounded"
      >
        <div
          class="MuiTableContainer-root container"
        >
          <table
            aria-label="sticky table"
            class="MuiTable-root MuiTable-stickyHeader"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  Value
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  Description
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-hover"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  myAccount
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  CHF
                   
                  100
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  This is a description.
                </td>
              </tr>
              <tr
                class="MuiTableRow-root MuiTableRow-hover"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  myAccount2
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  USD
                   
                  200
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
